Sprint_4

AgonianV opened the pull request.

AgonianV committed:
add 18 tests for Class TestBooksCollector

AgonianV committed:
add 19 tests and add in README test description

AlexKitIt commented on tests.py:
```diff
@@ -21,4 +22,110 @@ def test_add_new_book_add_two_books(self):
         assert len(collector.get_books_rating()) == 2
```
Нужно исправить: такого метода у тестируемой программы нет, он должен был быть удален

AlexKitIt commented on tests.py:
```diff
@@ -21,4 +22,110 @@ def test_add_new_book_add_two_books(self):
+        collectors.add_new_book('Луна')
+        collectors.set_book_genre('Луна', 'Фантастика')
+        collectors.add_book_in_favorites('Луна')
+        if len(collectors.favorites) == 2:
```
Нужно исправить: условий в тестах быть не может, тест всегда однозначен. Если это не вписывается в параметризацию, значит ее быть не должно и это просто несколько отдельных тестовых методов

AlexKitIt commented on tests.py:
```diff
@@ -21,4 +22,110 @@ def test_add_new_book_add_two_books(self):
+        collectors = BooksCollector()
+
+        collectors.add_new_book('Зона')
+        collectors.set_book_genre('Зона', 'Ужасы')
```
Для добавления в избранное этот шаг не нужен

AgonianV committed:
Выполнены исправления указанные в ревью:
1. Удален тест test_add_new_book_add_two_books
2. В тесте test_delete_book_from_favorites_delete_all_books убрано условие if
3. В тестах test_delete_book_from_favorites_delete_all_books и test_add_book_in_favorites_add_two_books убраны лишние шаги  "collectors.set_book_genre"
